tiledb/sm/query/ast/query_ast.cc

```diff
@@ -61,12 +61,31 @@ bool ASTNodeVal::is_backwards_compatible() const {
 
 Status ASTNodeVal::check_node_validity(const ArraySchema& array_schema) const {
   const uint64_t condition_value_size = condition_value_data_.size();
-  const auto attribute = array_schema.attribute(field_name_);
-  // Check that the field_name represents an attribute in the array
-  // schema corresponding to this QueryCondition object.
-  if (!attribute) {
-    return Status_QueryConditionError(
-        "Value node field name is not an attribute " + field_name_);
+  const bool is_dim = array_schema.is_dim(field_name_);
+
+  bool nullable = false;
```

Comment:
ArraySchema has `var_size`, `cell_size`, `cell_val_num`, and `type` accessors for getting the information below without conditional/access on attribute vs dimension.